component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/reporter/choreo/ChoreoInputValidator.java

```diff
@@ -0,0 +1,101 @@
+
+import static org.wso2.am.analytics.publisher.util.Constants.API_CREATION;
+import static org.wso2.am.analytics.publisher.util.Constants.API_CREATOR_TENANT_DOMAIN;
+import static org.wso2.am.analytics.publisher.util.Constants.API_ID;
+import static org.wso2.am.analytics.publisher.util.Constants.API_METHOD;
+import static org.wso2.am.analytics.publisher.util.Constants.API_NAME;
+import static org.wso2.am.analytics.publisher.util.Constants.API_RESOURCE_TEMPLATE;
+import static org.wso2.am.analytics.publisher.util.Constants.API_VERSION;
+import static org.wso2.am.analytics.publisher.util.Constants.APPLICATION_CONSUMER_KEY;
+import static org.wso2.am.analytics.publisher.util.Constants.APPLICATION_ID;
+import static org.wso2.am.analytics.publisher.util.Constants.APPLICATION_NAME;
+import static org.wso2.am.analytics.publisher.util.Constants.APPLICATION_OWNER;
+import static org.wso2.am.analytics.publisher.util.Constants.AUTH_API_URL;
+import static org.wso2.am.analytics.publisher.util.Constants.BACKEND_LATENCY;
+import static org.wso2.am.analytics.publisher.util.Constants.CONSUMER_KEY;
+import static org.wso2.am.analytics.publisher.util.Constants.CONSUMER_SECRET;
+import static org.wso2.am.analytics.publisher.util.Constants.CORRELATION_ID;
+import static org.wso2.am.analytics.publisher.util.Constants.DEPLOYMENT_ID;
+import static org.wso2.am.analytics.publisher.util.Constants.DESTINATION;
+import static org.wso2.am.analytics.publisher.util.Constants.ERROR_CODE;
+import static org.wso2.am.analytics.publisher.util.Constants.ERROR_MESSAGE;
+import static org.wso2.am.analytics.publisher.util.Constants.ERROR_TYPE;
+import static org.wso2.am.analytics.publisher.util.Constants.GATEWAY_TYPE;
+import static org.wso2.am.analytics.publisher.util.Constants.KEY_TYPE;
+import static org.wso2.am.analytics.publisher.util.Constants.PROXY_RESPONSE_CODE;
+import static org.wso2.am.analytics.publisher.util.Constants.REGION_ID;
+import static org.wso2.am.analytics.publisher.util.Constants.REQUEST_MEDIATION_LATENCY;
+import static org.wso2.am.analytics.publisher.util.Constants.RESPONSE_CACHE_HIT;
+import static org.wso2.am.analytics.publisher.util.Constants.RESPONSE_LATENCY;
+import static org.wso2.am.analytics.publisher.util.Constants.RESPONSE_MEDIATION_LATENCY;
+import static org.wso2.am.analytics.publisher.util.Constants.SAS_TOKEN;
+import static org.wso2.am.analytics.publisher.util.Constants.TARGET_RESPONSE_CODE;
+import static org.wso2.am.analytics.publisher.util.Constants.TOKEN_API_URL;
+import static org.wso2.am.analytics.publisher.util.Constants.USER_AGENT;
+
+/**
+ * Input Validator for {@link ChoreoAnalyticsMetricReporter}. Validator holds all required attributes against which
+ * inputs will be validated.
+ */
+public class ChoreoInputValidator {
+    private static final ChoreoInputValidator INSTANCE = new ChoreoInputValidator();
+    private static final String[] responseSchema = {CORRELATION_ID, KEY_TYPE, API_ID, API_NAME,
```

Comment:
are we going to support define schema types and schema from the config file later?

Reply:
schema will be dictated by avro in future.